miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,702 @@
+        """Set auto boost on carpet."""
+        return self.set_property("auto_boost", auto_boost)
+
+    def _set_dnd(self, start_int: int, end_int: int, active: bool):
+        value_str = json.dumps({"time": [start_int, end_int, int(active)]})
+        return self.set_property("forbid_mode", value_str)
+
+    @command(
+        click.argument("start_hr", type=int),
+        click.argument("start_min", type=int),
+        click.argument("end_hr", type=int),
+        click.argument("end_min", type=int),
+    )
+    def set_dnd(self, start_hr: int, start_min: int, end_hr: int, end_min: int):
+        """Set do-not-disturb.
+
+        :param int start_hr: Start hour
+        :param int start_min: Start minute
+        :param int end_hr: End hour
+        :param int end_min: End minute
+        """
+        start_int = int(timedelta(hours=start_hr, minutes=start_min).total_seconds())
+        end_int = int(timedelta(hours=end_hr, minutes=end_min).total_seconds())
+        return self._set_dnd(start_int, end_int, active=True)
+
+    @command()
+    def disable_dnd(self):
+        """Disable do-not-disturb."""
+        # The current do not disturb is read back for a better user expierence,
+        # as start/end time must be set together with enabled=False
+        try:
+            current_dnd_str = self.get_property_by(**_MAPPING["forbid_mode"])[0][
+                "value"
+            ]
+            current_dnd_dict = json.loads(current_dnd_str)
+        except Exception:
+            # In case reading current DND back fails, DND is disabled anyway
+            return self._set_dnd(0, 0, active=False)
+        return self._set_dnd(
+            current_dnd_dict["time"][0], current_dnd_dict["time"][1], active=False
+        )
+
+    @command(click.argument("water_level", type=EnumType(WaterLevel)))
+    def set_water_level(self, water_level: WaterLevel):
+        """Set water_level."""
+        return self.set_property("water_level", water_level.value)
+
+    @command(click.argument("double_clean", type=bool))
+    def set_double_clean(self, double_clean: bool):
+        """Set double clean (True/False)."""
+        return self.set_property("double_clean", double_clean)
+
+    @command(click.argument("edge_sweep", type=bool))
+    def set_edge_sweep(self, edge_sweep: bool):
+        """Set edge_sweep (True/False)."""
+        return self.set_property("edge_sweep", edge_sweep)
+
+    @command(click.argument("lidar_collision", type=bool))
+    def set_lidar_collision(self, lidar_collision: bool):
+        """Set lidar collision (True/False).."""
+        return self.set_property("lidar_collision", lidar_collision)
+
+    @command()
+    def start_dust(self) -> None:
+        """Start base dust collection."""
+        return self.call_action_by(8, 6)
+
+    @command(click.argument("voice", type=str))
+    def set_voice_unknown(self, voice: str) -> None:
+        """Set voice.
+
+        FIXME: the syntax of voice is unknown
+        """
```

Comment:
Could this be the name of the setting, e.g., "girl_en"?

Reply:
Yes, I think that "girl_en" can be part of the string. But I also think that it must be json format, and I do not know the json-key. (I have tried {"voice": "girl_en"} without lock)

Reply:
Okay, let's keep it this way, maybe someone figures it out at some point :-)